flink-runtime/src/test/java/org/apache/flink/runtime/rest/handler/legacy/backpressure/BackPressureRequestCoordinatorTest.java

```diff
@@ -0,0 +1,334 @@
+import org.apache.flink.runtime.executiongraph.ExecutionJobVertex;
+import org.apache.flink.runtime.executiongraph.ExecutionVertex;
+import org.apache.flink.runtime.executiongraph.IntermediateResult;
+import org.apache.flink.runtime.messages.TaskBackPressureResponse;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+/**
+ * Tests for the {@link BackPressureRequestCoordinator}.
+ */
+public class BackPressureRequestCoordinatorTest extends TestLogger {
+
+	private static final long requestTimeout = 10000;
+	private static final double backPressureRatio = 0.5;
+
+	private static ScheduledExecutorService executorService;
+	private BackPressureRequestCoordinator coordinator;
+
+	@BeforeClass
+	public static void setUp() throws Exception {
+		executorService = new ScheduledThreadPoolExecutor(1);
+	}
+
+	@AfterClass
+	public static void tearDown() throws Exception {
+		if (executorService != null) {
+			executorService.shutdown();
+		}
+	}
+
+	@Before
+	public void initCoordinator() throws Exception {
+		coordinator = new BackPressureRequestCoordinator(executorService, requestTimeout);
+	}
+
+	@After
+	public void shutdownCoordinator() throws Exception {
+		if (coordinator != null) {
+			coordinator.shutDown();
+		}
+	}
+
+	/** Tests simple request of task back pressure stats. */
+	@Test(timeout = 10000L)
+	public void testSuccessfulBackPressureRequest() throws Exception {
+		ExecutionVertex[] vertices = createExecutionVertices(ExecutionState.RUNNING, CompletionType.SUCCESSFULLY);
+
+		CompletableFuture<BackPressureStats> requestFuture = coordinator.triggerBackPressureRequest(vertices);
+		BackPressureStats backPressureStats = requestFuture.get();
+
+		// verify the request result
+		assertEquals(0, backPressureStats.getRequestId());
+		Map<ExecutionAttemptID, Double> backPressureRatios = backPressureStats.getBackPressureRatios();
+		for (ExecutionVertex executionVertex : vertices) {
+			ExecutionAttemptID executionId = executionVertex.getCurrentExecutionAttempt().getAttemptId();
+			assertEquals(backPressureRatio, backPressureRatios.get(executionId), 0.0);
+		}
+
+		// verify no more pending request
+		assertEquals(0, coordinator.getNumberOfPendingRequests());
```

Comment:
Migrate this assert into @Aftter `shutdownCoordinator()`, then all the tests can reuse it.